components/bl-category-chart/src/index.js

```diff
@@ -43,104 +17,166 @@ export default function CategoryChartComponent({ component, elRef }) {
+      },
+      title: {
+        display: titleVisibility,
+        text: title,
+        font: {
+          size: titleFontSize,
+        },
+      },
+    },
+    scales: {
+      y: {
+        grid: {
+          display: yGridLineVisibility,
+          color: gridLinesColor,
+          lineWidth: gridLinesWidth,
+        },
+      },
+      x: {
+        grid: {
+          display: xGridLineVisibility,
+          color: gridLinesColor,
+          lineWidth: gridLinesWidth,
+        },
+      },
+    },
+  }
 
   useEffect(() => {
     if (!display) {
-      return null;
+      return;
+    }
+
+    let requireRerender = false;
+
+    const optionsConfig = { ...options };
+
+    if (isDataChange(chartConfig, optionsConfig)) {
+      requireRerender = true;
+      Object.assign(chartConfig, optionsConfig);
+    }
+
+    const scaleConfig = {
+      y: {
+        grid: {
+          display: yGridLineVisibility,
+          color: gridLinesColor,
+          lineWidth: gridLinesWidth,
+        },
+      },
+      x: {
+        grid: {
+          display: xGridLineVisibility,
+          color: gridLinesColor,
+          lineWidth: gridLinesWidth,
+        },
+      },
+    };
+
+    if (isDataChange(chartConfig.scales, scaleConfig)) {
+      requireRerender = true;
+      Object.assign(chartConfig.scales.y, scaleConfig.y);
+      Object.assign(chartConfig.scales.x, scaleConfig.x);
     }
 
-    chartInstance.current = new Chart(chartRef.current, {
-      type,
-      data: { labels, datasets },
-      options: {
-        ...options,
-        plugins: {
-          title: {
-            display: chartTitleVisibility,
-            text: chartTitle,
-            font: {
-              size: chartTitleFontSize
-            }
-          }
+    const pluginsConfig = {
+      legend: {
+        display: legendVisibility,
+      },
+      title: {
+        display: titleVisibility,
+        text: title,
+        font: {
+          size: titleFontSize,
         },
-        scales: {
-          y: {
-            grid: {
-              display: yGridLineVisibility,
-              color: gridLinesColor,
-              lineWidth: gridLinesWidth
-            }
-          },
-          x: {
-            grid: {
-              display: xGridLineVisibility,
-              color: gridLinesColor,
-              lineWidth: gridLinesWidth
-            },
-          }
-        }
+      },
+    };
+
+    if (isDataChange(chartConfig.plugins, pluginsConfig)) {
+      requireRerender = true;
+      Object.assign(chartConfig.plugins.legend, pluginsConfig.legend);
+      Object.assign(chartConfig.plugins.title, pluginsConfig.title);
+    }
+
+    if (chartInstance.current && requireRerender) {
+      chartInstance.current.data = { labels, datasets };
+      chartInstance.current.update();
+    } else {
+      chartInstance.current = new Chart(chartRef.current, {
+        type,
+        data: { labels, datasets },
+        options: chartConfig,
+      });
```

Comment:
Should it be in separate useEffect?

```
useEffect(()=> {
  chartInstance.current = new Chart(chartRef.current, {
    type,
    data: { labels, datasets },
    options: chartConfig,
  });
  
  return () => {
    chartInstance.current.destroy();
    chartInstance.current = null;
  }
}, [])

Reply:
They can be separated, but in the previous version we tried to avoid this.

Reply:
Avoid what?